model.go

```diff
@@ -352,27 +294,31 @@ func (m *Model) unsatisfiedMachineAndUnitPlacements(sourceApp string, placements
 }
 
 func (m *Model) machineHasApp(machine, appName, containerType string) bool {
+	return m.machineHasAppUnit(machine, appName, containerType) != ""
+}
+
+func (m *Model) machineHasAppUnit(machine, appName, containerType string) string {
```

Comment:
This should be called machineAppUnit() or getMachineAppUnit() since that's what it does.

Channelling William, it should really return (string, bool)